mlir/lib/Rewrite/PatternApplicator.cpp

```diff
@@ -40,7 +40,9 @@ static void logImpossibleToMatch(const Pattern &pattern) {
 
 /// Log IR after pattern application.
 static Operation *getDumpRootOp(Operation *op) {
-  return op->getParentWithTrait<mlir::OpTrait::IsIsolatedFromAbove>();
+  return op->hasTrait<mlir::OpTrait::IsIsolatedFromAbove>()
+             ? op
+             : op->getParentWithTrait<mlir::OpTrait::IsIsolatedFromAbove>();
```

Comment:
If you wish to fix the nullptr, I would likely write it as:

```suggestion
  Operation *isolatedParent = op->getParentWithTrait<mlir::OpTrait::IsIsolatedFromAbove>();
  if (isolatedParent) return isolatedParent;
  return op;
```

Reply:
This change makes sense to me. If there is no isolated-from-above parent, we are modifying the top-level module and we should dump it directly.

Reply:
You are right, this is better. I have pushed an update